### PR TITLE
Lang mapper ui

### PR DESCRIPTION
**Pull request by phlax**
Adds a ui for configuring pootle <> fs language mappings

### CODE REVIEW

File: pootle/templates/admin/project_fs.html [hunk 1]
```diff
@@ -15,35 +15,70 @@
...
+	    {% blocktrans %}For other backends the translation path will be relative to the root of the repository.{% endblocktrans %}
+	  </p>
+	</form>
+      </div>
     </div>
     <div class="config-content">
-      Language mapping configuration
-      
+      <div class="config-content-panel">
+	<h3>{% blocktrans with project_code=project.code %}Public key{% endblocktrans %}</h3>
```
Comment (unho) [inline pootle/templates/admin/project_fs.html]: The `project_code` is not needed.

Comment (unho) [inline pootle/templates/admin/project_fs.html]: Same below.

File: pootle/templates/admin/project_fs.html [hunk 2]
```diff
@@ -15,35 +15,70 @@
...
+	  {% csrf_token %}
+	  {{ lang_mapping_formset.management_form }}
+	  {% for error in lang_mapping_formset.non_form_errors %}
+	  <div class="error" style="color: red; font-weight: bold; padding: 1em 0">{{ error }}</div>
+	  {% endfor %}
+	  <table>
+	    <tr>
+	      <th style="text-align: right">Remove</th>
+	      <th>Pootle code</th>
+	      <th>Filesystem code</th>
```
Comment (unho) [inline pootle/templates/admin/project_fs.html]: Make these messages translatable.

File: pootle/apps/pootle_fs/forms.py [hunk 3]
```diff
@@ -90,3 +95,56 @@ def save(self):
...
+
+class LangMappingForm(forms.Form):
+    remove = forms.BooleanField(required=False)
+    pootle_code = forms.ModelChoiceField(
+        Language.objects.all(),
+        to_field_name="code",
+        widget=forms.Select(
+            attrs={'class': 'js-select2'}))
+    fs_code = forms.CharField(
+        label=_("Filesystem code"), max_length=32)
```
Comment (unho) [inline pootle/apps/pootle_fs/forms.py]: I don't think the label is needed with current UI.

File: pootle/templates/admin/project_fs.html [hunk 4]
```diff
@@ -15,35 +15,70 @@
...
+	<code>{{ public_key }}</code>
+      </div>
+      <div class="config-content-panel">
+	<h3>{% blocktrans with project_code=project.code %}Language mapping configuration{% endblocktrans %}</h3>
+
+	<form method="post" action___="{{ lang_mapping_formset_action }}">
+	  {% csrf_token %}
+	  {{ lang_mapping_formset.management_form }}
+	  {% for error in lang_mapping_formset.non_form_errors %}
+	  <div class="error" style="color: red; font-weight: bold; padding: 1em 0">{{ error }}</div>
```
Comment (unho) [inline pootle/templates/admin/project_fs.html]: The styling should be not in the template.

File: pootle/apps/pootle_fs/views.py [hunk 5]
```diff
@@ -6,12 +6,17 @@
 # or later license. See the LICENSE file for a copy of the license and the
 # AUTHORS file for copyright and authorship information.
 
+import getpass
+import os
+import pwd
+
+
```
Comment (unho) [inline pootle/apps/pootle_fs/views.py]: Only one blank line between imports.

File: pootle/templates/admin/project_fs.html (hunk 4 above)
Comment (phlax) [inline pootle/templates/admin/project_fs.html]: this needs to be fixed...

File: pootle/apps/pootle_fs/forms.py (hunk 3 above)
Comment (phlax) [inline pootle/apps/pootle_fs/forms.py]: removing..

File: pootle/templates/admin/project_fs.html [hunk 6]
```diff
@@ -15,35 +15,70 @@
...
+	  {% endfor %}
+	  <table>
+	    <tr>
+	      <th style="text-align: right">Remove</th>
+	      <th>Pootle code</th>
+	      <th>Filesystem code</th>
+	    </tr>
+	    {% for form in lang_mapping_formset %}
+	    <tr>
+	      <td style="text-align: right; padding-right: .5em">{% if form.initial %}{{ form.remove }}{% else %}Add{% endif %}</td>
```
Comment (phlax) [inline pootle/templates/admin/project_fs.html]: im happy with this ui for now - we can/will revisit these widgets soon

File: pootle/templates/admin/project_fs.html [hunk 7]
```diff
@@ -15,35 +15,66 @@
...
+	      <code>
+		`/path/to/translation/files/&lt;language_code&gt;/&lt;dir_path&gt;/&lt;filename&gt;.&lt;ext&gt;`
+	      </code>
+	    </blockquote>
+	  </p>
+	  <p>
+	    {% blocktrans %}Only `&lt;language_code&gt;` is required, although it must end with `.&lt;ext&gt;`{% endblocktrans %}
+	  </p>
+	  <p>
+	    {% blocktrans %}If you are using the localfs backend the translation path will be combined with the path specified in
```
Comment (unho) [inline pootle/templates/admin/project_fs.html]: Use `trimmed` in `blocktrans`.

File: pootle/templates/admin/project_fs.html [hunk 8]
```diff
@@ -15,35 +15,66 @@
...
+	    {% for form in lang_mapping_formset %}
+	    <tr>
+	      <td class="aligned-right padding-half-right">{% if form.initial %}{{ form.remove }}{% else %}Add{% endif %}</td>
+	      <td>{{ form.pootle_code }} {{ form.pootle_code.errors }}</td>
+	      <td>{{ form.fs_code }}</td>
+	    </tr>
+	    {% endfor %}
+	    <tr>
+	      <td colspan="3" class="centered">
+		<input type="submit" class="btn" value="{% trans Update %}" />
```
Comment (unho) [inline pootle/templates/admin/project_fs.html]: `Update` var doesn't exist, so enclose translatable text in quotes.

File: pootle/templates/admin/project_fs.html [hunk 9]
```diff
@@ -15,35 +15,66 @@
...
+	  {% endfor %}
+	  <table>
+	    <tr>
+	      <th class="aligned-right">{% blocktrans %}Remove{% endblocktrans %}</th>
+	      <th>{% blocktrans %}Pootle code{% endblocktrans %}</th>
+	      <th>{% blocktrans %}Filesystem code{% endblocktrans %}</th>
+	    </tr>
+	    {% for form in lang_mapping_formset %}
+	    <tr>
+	      <td class="aligned-right padding-half-right">{% if form.initial %}{{ form.remove }}{% else %}Add{% endif %}</td>
```
Comment (unho) [inline pootle/templates/admin/project_fs.html]: "Add" no translatable?

File: pootle/templates/admin/project_fs.html (hunk 8 above)
Comment (phlax) [inline pootle/templates/admin/project_fs.html]: fixed already 8/

**Comment by unho** [comment]
lgtm despite not all my concerns addressed.

**Pull request merged**